### PR TITLE
test(daemon): expand abbenay-service and consumer-auth unit coverage

### PR DESCRIPTION
**Pull request by sudhirverma**
## Summary

- Expand `abbenay-service.test.ts` with gRPC handler coverage (auth, streaming, MCP, config, sessions, providers)
- Add `consumer-auth` edge-case tests (env variants, token matching, missing env values)
- Remove tautological SessionChat simulations and `authorizeConsumer` spies; use real auth paths and `withEnv` cleanup
- Address CodeRabbit review: deterministic env in consumer-auth tests, shared `withEnv` helper, fake-timer `try/finally`, handler assertion coverage

## Commits

- `test(daemon): expand abbenay-service and consumer-auth unit coverage` — comprehensive handler tests and consumer-auth edge cases
- `test(daemon): address CodeRabbit review on daemon server tests` — explicit env maps, shared `withEnv`, timer cleanup, camelCase handler assertions

## Coverage

| File | Lines | Statements | Branches | Functions |
|------|-------|------------|----------|-----------|
| `abbenay-service.ts` | 99.4% | 96.7% | 83% | 100% |
| `consumer-auth.ts` | 100% | 100% | 98% | 100% |

## Test plan

- [x] `npx eslint src/daemon/server/abbenay-service.test.ts src/daemon/server/consumer-auth.test.ts src/daemon/server/test-env.ts`
- [x] `npx vitest run src/daemon/server/abbenay-service.test.ts src/daemon/server/consumer-auth.test.ts`
- [ ] `npm test` (full monorepo)
- [ ] `npm run ci:build`

### CODE REVIEW

File: packages/daemon/src/daemon/server/consumer-auth.test.ts [hunk 1]
```diff
@@ -107,6 +107,18 @@ describe('hasConfiguredConsumers / resolveAllowOpenAuth / buildConsumerAuthConte
       grpcHost: '127.0.0.1',
     })).toEqual({ loopbackOnly: true, allowOpenAuth: false });
   });
+
+  it('defaults grpcHost to loopback when grpcPort set without host', () => {
+    expect(buildConsumerAuthContext({ grpcPort: 50051 })).toEqual({
+      loopbackOnly: true,
+      allowOpenAuth: false,
+    });
```
[the remark this answers is not in the document]
Comment (sudhirverma) [reply packages/daemon/src/daemon/server/consumer-auth.test.ts:107]: Fixed in a1d4e66. `buildConsumerAuthContext` tests pass explicit `env: {}`. `matchConsumerByToken` tests use explicit env maps instead of `process.env`. `afterEach` also clears `NOT_SET_ENV` and `ABBENAY_ALLOW_OPEN_AUTH`.

**Comment by sudhirverma** [comment]
Addressed CodeRabbit review feedback in a1d4e66:

- **Environment isolation** (`consumer-auth.test.ts`): explicit `env` maps for `buildConsumerAuthContext` and `matchConsumerByToken`; `afterEach` clears `NOT_SET_ENV` and `ABBENAY_ALLOW_OPEN_AUTH`.
- **Fake timer cleanup** (`abbenay-service.test.ts`): `Shutdown` and both `VSCodeStream` tests use `try/finally` so timers and spies always restore.
- **Shared `withEnv`**: extracted to `packages/daemon/src/daemon/server/test-env.ts`.
- **Coverage-only handler calls**: added error and mock assertions for camelCase Register/ListModels/SummarizeSession/GetSession/ListSessions/DeleteSession/ReconnectMcpServer/ConfigureProvider/RemoveProvider invocations.

**Review by sathyapramod — APPROVED** [review]
LGTM!!

**Review by sathyapramod — APPROVED** [review]

**Review by Hrithik-Gavankar — APPROVED** [review]
